dataflow/src/main/java/org/checkerframework/dataflow/cfg/ControlFlowGraph.java

```diff
@@ -330,4 +364,21 @@ public String toStringDebug() {
         result.add("declaredLambdas=" + declaredLambdas);
         return result.toString();
     }
+
+    /** Checks representation invariants on this. */
+    public void checkRep(@UnknownInitialization(ControlFlowGraph.class) ControlFlowGraph this) {
+        List<Node> allNodes = getAllNodes();
+
+        // Require that each node in treeLookup exists in this ControlFlowGraph
```

Comment:
This is a new invariant. It's useful to know to which nodes a tree maps to even if the tree is in dead code.  (See the other comment for an example.)  Can you explain more why this new invariant is worth the loss of information?

Reply:
It would be less expensive to precompute the information once, if in the future clients want to know whether an expression is dead.

But, if you think the change is a bad idea we can drop the pull request.  (There should still be some kind of documentation for this data structure, which I found underdocumented and confusing.)